packages/bridge/src/skip/index.ts

```diff
@@ -621,7 +614,77 @@ export class SkipBridgeProvider implements BridgeProvider {
     return addressList;
   }
 
-  getFinalityTimeForChain(chainID: string) {
+  /**
+   * Sums the total transfer time of each hop between chains
+   * @returns total transfer time in seconds
+   */
+  async estimateTotalTransferTime(chainIds: string[]): Promise<number> {
+    if (chainIds.length < 2) {
+      throw new Error(
+        "At least two chain IDs are required to estimate transfer time."
+      );
+    }
+
+    let totalTransferTime = 0;
+
+    for (let i = 0; i < chainIds.length - 1; i++) {
+      const fromChainId = chainIds[i];
+      const toChainId = chainIds[i + 1];
+      const transferTime = await this.estimateTransferTime(
+        fromChainId,
+        toChainId
+      );
+      totalTransferTime += transferTime;
+    }
+
+    return totalTransferTime;
+  }
+
+  /**
+   * Estimates the transfer time for IBC transfers in seconds.
+   * Looks at the average block time of the two chains.
+   * @returns transfer time in seconds
+   */
+  async estimateTransferTime(
+    fromChainId: string,
+    toChainId: string
+  ): Promise<number> {
+    const fromChain = this.ctx.chainList.find(
+      (c) => c.chain_id === fromChainId
+    );
+    const toChain = this.ctx.chainList.find((c) => c.chain_id === toChainId);
+
+    const fromRpc = fromChain?.apis.rpc[0]?.address;
+    const toRpc = toChain?.apis.rpc[0]?.address;
+
+    const [fromBlockTimeMs, toBlockTimeMs] = await Promise.all([
+      fromRpc
+        ? queryRPCStatus({ restUrl: fromRpc }).then(calcAverageBlockTimeMs)
+        : this.getFinalityTimeForEvmChain(fromChainId) * 1000,
+      toRpc
+        ? queryRPCStatus({ restUrl: toRpc }).then(calcAverageBlockTimeMs)
+        : this.getFinalityTimeForEvmChain(toChainId) * 1000,
+    ]);
```

Comment:
If we're unsure whether the RPC address is defined, we should handle the case where it's a Cosmos chain and the address is undefined. Otherwise, we could send the wrong chain id to getFinalityTimeForEvmChain.

Reply:
In that case perhaps we can use a hardcoded value? Like 7.5 seconds for a cosmos chain

Reply:
But also, an RPC should be defined with pretty high confidence in the chain registry though

Reply:
I agree that this is unlikely to occur, however for insurance a 7.5 seconds like you mention could be a good fallback